sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt

```diff
@@ -0,0 +1,627 @@
+            tileRegion = region
+          }
+        }
+        if (it.isError) {
+          Logger.e(TAG, "Error: ${it.error}")
+          it.error?.let { error ->
+            tileRegionError = error
+          }
+        }
+        latch.countDown()
+      }
+    }
+    if (!latch.await(30, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    } else {
+      Assert.assertNotNull(tileRegion)
+      Assert.assertNull(tileRegionError)
+      Assert.assertNotNull(tileRegionProgress)
+
+      Assert.assertEquals(
+        tileRegion?.requiredResourceCount,
+        tileRegion?.completedResourceCount
+      )
+      Assert.assertEquals(
+        tileRegionProgress?.requiredResourceCount,
+        tileRegionProgress?.completedResourceCount
+      )
+      Assert.assertEquals(0L, tileRegionProgress?.erroredResourceCount)
+    }
+  }
+
+  @Test
+  fun loadTileRegionCancel() {
+    val latch = CountDownLatch(1)
+    var tileRegionProgress: TileRegionLoadProgress? = null
+    var tileRegion: TileRegion? = null
+    var tileRegionError: TileRegionError? = null
+    var cancelableTileStoreTask: Cancelable? = null
+    handler.post {
+      cancelableTileStoreTask = TileStore.getInstance().loadTileRegion(
+        TILE_REGION_ID,
+        TileRegionLoadOptions.Builder()
+          .geometry(TOKYO)
+          .descriptors(
+            listOf(
+              offlineManager.createTilesetDescriptor(
+                TilesetDescriptorOptions.Builder()
+                  .styleURI(STYLE)
+                  .minZoom(ZOOM.toInt().toByte())
+                  .maxZoom(ZOOM.toInt().toByte())
+                  .build()
+              )
+            )
+          )
+          .metadata(Value(TILE_REGION_METADATA))
+          .tileLoadOptions(
+            TileLoadOptions.Builder()
+              .criticalPriority(false)
+              .acceptExpired(true)
+              .networkRestriction(NetworkRestriction.NONE)
+              .build()
+          )
+          .build(),
+        {
+          tileRegionProgress = it
+          cancelableTileStoreTask?.cancel()
+        }
+      ) {
+        if (it.isValue) {
+          it.value?.let { region ->
+            tileRegion = region
+          }
+        }
+        if (it.isError) {
+          Logger.e(TAG, "Error: ${it.error}")
+          it.error?.let { error ->
+            tileRegionError = error
+          }
+        }
+        latch.countDown()
+      }
+    }
+    if (!latch.await(30, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    } else {
+      Assert.assertNull(tileRegion)
+      Assert.assertNotNull(tileRegionError)
+      Assert.assertNotNull(tileRegionProgress)
+
+      Assert.assertNotEquals(
+        tileRegionProgress?.requiredResourceCount,
+        tileRegionProgress?.completedResourceCount
+      )
+      Assert.assertEquals(TileRegionErrorType.CANCELED, tileRegionError?.type)
+    }
+  }
+
+  // TODO revisit after https://github.com/mapbox/mapbox-sdk-common/issues/1311
```

Comment:
nit, remove links to internal issue.

Reply:
@tobrun aren't we allowed to add such links in code?

Reply:
idk anymore :) we started off with never using any links to internal and now we are so :shrug:

Reply:
looked through code, indeed we do not have such internal links anymore, will remove it and create a new task.